clients/gateway/gateway.go

```diff
@@ -57,8 +57,8 @@ func newTestServer() *httptest.Server {
 			return
 		}
 
-		txHash := new(felt.Felt).SetBytes([]byte("random"))
-		resp := fmt.Sprintf("{\"code\": \"TRANSACTION_RECEIVED\", \"transaction_hash\": %q}", txHash.String())
+		hash := new(felt.Felt).SetBytes([]byte("random"))
+		resp := fmt.Sprintf("{\"code\": \"TRANSACTION_RECEIVED\", \"transaction_hash\": %q, \"address\": %q}", hash.String(), hash.String())
```

Comment:
Sure! A successful `addInvokeTransaction` method only needs to return a `transaction hash`, whereas the `addDeployAccountTransaction` method requires a` transaction hash` and `contract address`. I just return both here (even for the `addInvokeTransaction`) because we're just using this for test purposes, and it keeps the code simple. The gateway also returns a "code" of type string, which isn't apart of the spec, but I included it anyway because we're mocking the gateways response.